gnomad_qc/v3/create_release/sanity_checks.py

```diff
@@ -0,0 +1,711 @@
+            if value:
+                values_found = t_y.aggregate(
+                    hl.agg.filter(
+                        hl.is_defined(t_y.info[metric]),
+                        hl.agg.take(t_y.info[metric], 1),
+                    )
+                )
+                logger.info(
+                    f"FAILED {metric} = {None} check for Y variants. Values found: {values_found}"
+                )
+            else:
+                logger.info(f"PASSED {metric} = {None} check for Y variants")
+
+    t_x = hl.filter_intervals(t, [hl.parse_locus_interval("chrX")])
+    t_xnonpar = t_x.filter(t_x.locus.in_x_nonpar())
+    n = t_xnonpar.count()
+    logger.info(f"Found {n} X nonpar sites")
+
+    logger.info("Check (nhomalt == nhomalt_female) for X nonpar variants:")
+    female_metrics = [x for x in female_metrics if "nhomalt" in x]
+    for metric in female_metrics:
+        standard_field = metric.replace("-female", "").replace("-XX", "")
+        generic_field_check(
+            t_xnonpar,
+            (t_xnonpar.info[f"{metric}"] != t_xnonpar.info[f"{standard_field}"]),
+            f"{metric} == {standard_field}",
+            [f"info.{metric}", f"info.{standard_field}"],
+            verbose,
+        )
+
+
+def missingness_sanity_checks(
+    t: Union[hl.MatrixTable, hl.Table],
+    info_metrics: List[str],
+    non_info_metrics: List[str],
+    n_sites: int,
+    missingness_threshold: float,
+) -> None:
+    """
+    Check amount of missingness in all row annotations.
+    Prints metric to terminal if more than missingness_threshold% of annotations for that metric are missing.
+    :param hl.MatrixTable, hl.Table t: Input MatrixTable or Table.
+    :param List[str] info_metrics: List of metrics in info struct of input Table.
+    :param List[str] non_info_metrics: List of row annotations minus info struct from input Table.
+    :param int n_sites: Number of sites in input Table.
+    :param float missingness_threshold: Upper cutoff for allowed amount of missingness.
+    :return: None
+    :rtype: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    logger.info(
+        f"Missingness threshold (upper cutoff for what is allowed for missingness checks): {missingness_threshold}"
+    )
+    metrics_frac_missing = {}
+    for x in info_metrics:
+        metrics_frac_missing[x] = hl.agg.sum(hl.is_missing(t.info[x])) / n_sites
+    for x in non_info_metrics:
+        metrics_frac_missing[x] = hl.agg.sum(hl.is_missing(t[x])) / n_sites
+    output = t.aggregate(hl.struct(**metrics_frac_missing))
+
+    n_fail = 0
+    for metric, value in dict(output).items():
+        message = f"missingness check for {metric}: {100 * value}% missing"
+        if value > missingness_threshold:
+            logger.info(f"FAILED {message}")
+            n_fail += 1
+        else:
+            logger.info(f"Passed {message}")
+    logger.info(f"{n_fail} missing metrics checks failed")
+
+
+def vcf_field_check(
```

Comment:
so sorry, again I'm not sure of the full overlap between UKBB and gnomad, but if this is something we consistently want to use for a sanity check in QC repos we should probably move to the common code